packages/fast-components-react-base/src/checkbox/checkbox.tsx

```diff
@@ -87,6 +87,62 @@ class Checkbox extends Foundation<ICheckboxHandledProps & ICheckboxManagedClasse
 
         return super.generateClassNames(classes);
     }
+
+    /**
+     * Generates the correct attributes
+     */
+    private generateAttributes(): object {
+        const attributes: object = {};
+
+        /* tslint:disable:no-string-literal */
+        if (this.props.disabled) {
+            attributes["disabled"] = "disabled";
+        }
+
+        // If the author is not setting the checked state, we should let the checkbox manage itself
+        if (!isUndefined(this.props.checked)) {
+            if (this.props.onChange) {
```

Comment:
It seems unintuitive to me to respect the `checked` prop *only* when `onChange` is passed. As an author, I would expect passing `checked` would *always* set the checked property, regardless of if I'm handling the change callback.